modules/genomic_browser/ajax/genomic_file_upload.php

```diff
@@ -51,16 +52,22 @@
                 'description'       => $_POST['description'],
                );
 
+setFullPath($fileToUpload);
+
 switch ($fileToUpload->genomic_file_type) {
 case 'Methylation beta-values':
     validateRequest();
-    moveFileToFS($fileToUpload);
+    begin();
```

Comment:
I find the way this is written misleading, in that it's looks like it does no error handling and always ends with success even if something goes wrong. (In reality, it has die statements hidden throughout the code and depends heavily on side-effects.)

Can this logic be rewritten to use exceptions and try/catch instead? That would make it obvious where the failure is handled, without significantly complicating the happy path.